### PR TITLE
[Android] fixes inconsistencies in settings

### PR DESCRIPTION
**Pull request by alexsafe**
Resolves 
https://github.com/brave/brave-browser/issues/16298
https://github.com/brave/brave-browser/issues/16304
https://github.com/brave/brave-browser/issues/16297
## Submitter Checklist:

- [x ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by deeppandya — APPROVED** [review]
LGTM

File: android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java [hunk 1]
```diff
@@ -335,58 +327,59 @@ private void updatePreferences() {
...
+        int fingerprintingPref = sharedPreferences.getInt(PREF_FINGERPRINTING_PROTECTION, 1);
+
         if (getNoScriptControlType.equals(BraveShieldsContentSettings.BLOCK_RESOURCE)) {
             mBlockScriptsPref.setChecked(true);
         } else {
             mBlockScriptsPref.setChecked(false);
         }
 
-        if (cosmeticFiltering.equals(BraveShieldsContentSettings.BLOCK_THIRDPARTY_RESOURCE)) {
+        if (blockAdTrackersPref == 0) {
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java:346]: It's not obvious what these options (0, 1, 2) really mean. We should probably explicitly define them.

Comment (alexsafe) [reply android/java/org/chromium/chrome/browser/privacy/settings/BravePrivacySettings.java:346]: not sure if duplicates, but they use they use the code in there to hook up to the browser implementation

**Review by samartnik — APPROVED** [review]
++

**Comment by srirambv** [comment]
Verification passed on OnePlus 6T with Android 10 running 1.27.59 x64 Nightly build
- Verified Ads/trackers is set to standard by default
- Verified changing one global setting doesn't affect others
- Verified when Brave shields & privacy is not reordered when loading the setting


https://user-images.githubusercontent.com/17010094/121869636-2768c500-ccfa-11eb-8412-45856f75dcc9.mp4

**Pull request merged**